sunbeam_migrate/tests/integration/handlers/neutron/test_router.py

```diff
@@ -0,0 +1,241 @@
+    )
+    request.addfinalizer(
+        lambda: test_source_session.network.remove_interface_from_router(
+            router.id, subnet_id=internal_subnet.id
+        )
+    )
+
+    # Migrate router with both dependencies and members
+    test_utils.call_migrate(
+        test_config_path,
+        [
+            "start",
+            "--resource-type=router",
+            "--include-dependencies",
+            "--include-members",
+            router.id,
+        ],
+    )
+
+    # Verify migrated router
+    dest_router = test_destination_session.network.find_router(router.name)
+    assert dest_router, "couldn't find migrated router"
+
+    _check_migrated_router(router, dest_router)
+
+    # Verify external gateway is set up correctly
+    dest_external_gateway_info = getattr(dest_router, "external_gateway_info", None)
+    assert dest_external_gateway_info, "external gateway info not migrated"
+
+    dest_external_network_id = dest_external_gateway_info.get("network_id")
+    assert dest_external_network_id, "external network id not set"
+
+    dest_external_network = test_destination_session.network.get_network(
+        dest_external_network_id
+    )
+
+    # Verify external subnet was migrated as dependency
+    dest_external_fixed_ips = dest_external_gateway_info.get("external_fixed_ips", [])
+    assert dest_external_fixed_ips, "external fixed IPs not migrated"
+
+    dest_external_subnet_id = dest_external_fixed_ips[0].get("subnet_id")
+    assert dest_external_subnet_id, "external subnet id not set"
+
+    dest_external_subnet = test_destination_session.network.get_subnet(
+        dest_external_subnet_id
+    )
+
+    # Verify internal subnet was migrated as member
+    dest_internal_subnet_id = test_utils.get_destination_resource_id(
+        test_config_path, "subnet", internal_subnet.id
+    )
+    dest_internal_subnet = test_destination_session.network.get_subnet(
+        dest_internal_subnet_id
+    )
+    assert dest_internal_subnet, "internal subnet not migrated as member"
+
+    # Verify the internal subnet is attached to the router
+    dest_ports = list(
+        test_destination_session.network.ports(
+            device_id=dest_router.id, device_owner="network:router_interface"
+        )
+    )
+    assert dest_ports, "no ports attached to migrated router"
+
+    # Check if the subnet is attached through any port
+    attached_subnet_ids = set()
+    for port in dest_ports:
+        for fixed_ip in getattr(port, "fixed_ips", []):
+            subnet_id = fixed_ip.get("subnet_id")
+            if subnet_id:
+                attached_subnet_ids.add(subnet_id)
+
+    assert dest_internal_subnet_id in attached_subnet_ids, (
+        "internal subnet not attached to router"
+    )
+
+    # Cleanup: delete the migrated internal network and subnet
+    dest_internal_network_id = test_utils.get_destination_resource_id(
+        test_config_path, "network", internal_network.id
+    )
+
+    request.addfinalizer(
```

Comment:
I won't block the PR for this, but the finalizers should be defined right after the resource gets created or the id is obtained. 

If one of the assertions above fails, than the resources will be leaked.